db/compaction/compaction.h

```diff
@@ -281,6 +281,8 @@ class Compaction {
 
   int GetInputBaseLevel() const;
 
+  const char* GetCompactionReasonString() const;
```

Comment:
In order to be publicly visible (i.e., usable by RocksDB users), the function needs to be declared in a header file under "include/" directory. For example take a look at how `GetStringFromCompressionType()` is declared: https://github.com/facebook/rocksdb/blob/e171a219d592b1bda9b654db126f04d4d834a815/include/rocksdb/convenience.h#L296-L297.

Reply:
Sorry I probably shouldn't have provided this example here, as the declaration itself is not substantial. Meanwhile, the example function looks too similar to what our function does so it's tempting to follow its implementation pattern.